### PR TITLE
Allow for use of Data.Text with twelve-days exercise

### PR DESCRIPTION
**Pull request by barrymoo**
This allows students to use `Data.Text` to solve the `twelve-days` exercise.

See #841

### CODE REVIEW

File: exercises/twelve-days/test/Tests.hs [hunk 1]
```diff
@@ -14,7 +15,7 @@ specs = describe "responseFor" $ for_ cases test
   where
       test Case{..} = it description assertion
         where
-          assertion = recite start stop `shouldBe` expected
+          assertion = convertString <$> recite start stop `shouldBe` expected
```
Comment (sshine) [inline exercises/twelve-days/test/Tests.hs]: we can just use this here:

```suggestion
          assertion = recite start stop `shouldBe` fromString expected
```

File: exercises/twelve-days/test/Tests.hs [hunk 2]
```diff
@@ -1,8 +1,9 @@
 {-# LANGUAGE RecordWildCards #-}
 
-import Data.Foldable     (for_)
-import Test.Hspec        (Spec, describe, it, shouldBe)
-import Test.Hspec.Runner (configFastFail, defaultConfig, hspecWith)
+import Data.Foldable           (for_)
+import Test.Hspec              (Spec, describe, it, shouldBe)
+import Test.Hspec.Runner       (configFastFail, defaultConfig, hspecWith)
+import Data.String.Conversions (convertString)
```
Comment (sshine) [inline exercises/twelve-days/test/Tests.hs]: These changes can be reverted

File: exercises/twelve-days/README.md [hunk 3]
```diff
@@ -28,6 +28,46 @@ On the eleventh day of Christmas my true love gave to me: eleven Pipers Piping,
...
 
+## Hints
+
+You need to implement the `recite` function which outputs the lyrics to
+'The Twelve Days of Christmas'. You can use the provided signature
+if you are unsure about the types, but don't let it restrict your creativity.
+
+```haskell
+recite :: Int -> Int -> [String]
+```
```
Comment (sshine) [inline exercises/twelve-days/README.md]: Perhaps remove this:

The `Text` alternative is provided below, and this type signature already occurs in the stub file.

File: exercises/twelve-days/examples/success-standard/package.yaml [hunk 4]
```diff
@@ -14,3 +14,4 @@ tests:
     dependencies:
       - twelve-days
       - hspec
+      - string-conversions
```
Comment (sshine) [inline exercises/twelve-days/examples/success-standard/package.yaml]:
```suggestion
```

File: exercises/twelve-days/package.yaml [hunk 5]
```diff
@@ -19,3 +19,4 @@ tests:
     dependencies:
       - twelve-days
       - hspec
+      - string-conversions
```
Comment (sshine) [inline exercises/twelve-days/package.yaml]:
```suggestion
```

File: exercises/twelve-days/test/Tests.hs (hunk 1 above)
Comment (barrymoo) [reply exercises/twelve-days/test/Tests.hs]: This makes a lot of sense.

File: exercises/twelve-days/README.md (hunk 3 above)
Comment (barrymoo) [reply exercises/twelve-days/README.md]: I made this change and also tried to clean up the unordered list phrases. Each entry in the list is an extension of "you can". I hope it reads a bit more cleanly now.

**Comment by barrymoo** [comment]
I squashed my first commit as well.

**Pull request merged**